lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+            }
+            return make_response(jsonify(response_object), 201)
+
+    @Auth.token_auth.login_required(optional=True)
+    def get(self, key):
+        """
+        Get request for querying all the metadata sessions for a user.
+        returns the list of all the metadata sessions associated with a logged in user.
+        This requires a Pbench auth token in the header field
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                    "data": {
+                        "sessions": [
+                            {"id": "metadata_id",
+                            "value": "client text blob",
+                            "key": "key string",
+                            "created": "datetime string",
+                            "updated": "datetime string", },
+                            {}]
+                        }
+                }
+        """
+        if not key:
+            self.logger.warning("Metadata key not provided during metadata query")
+            abort(400, message="Missing metadata key in the query url")
+
+        current_user = self.auth.token_auth.current_user()
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+        print(current_user_id)
+        try:
+            # Fetch the metadata sessions
+            # If the key is favorite, we return only favorite sessions,
+            # else we return all the saved and favorite sessions
+            if key.upper() == "FAVORITE":
+                metadata_sessions = Metadata.query(
+                    user_id=current_user_id, key=MetadataKeys[key.upper()].value
+                )
+            else:
+                metadata_sessions = Metadata.query(user_id=current_user_id)
+            data = [session.get_json() for session in metadata_sessions]
+        except Exception:
+            self.logger.exception(
+                "Exception occurred while querying the Metadata model"
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        response_object = {
+            "message": "success",
+            "data": {"sessions": data},
+        }
+        return make_response(jsonify(response_object), 200)
+
+
+class QueryMetadata(Resource):
+    """
+    Abstracted pbench API for querying a single user metadata session
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+
+    def verify_metadata(self, session, action):
+        current_user_id = Auth.token_auth.current_user().id
+        metadata_user_id = session.user_id
+        if current_user_id != metadata_user_id:
+            self.logger.warning(
+                "Metadata {}: Logged in user_id {} is different than the one provided in the URI {}",
```

Comment:
Technically the URI doesn't provide a "user_id". I think this also confirms to an "attacker" that their metadata row ID was valid but belongs to someone else. Let's just say that the metadata id they provided isn't valid.

Reply:
@Dave, this is for the log, not the response to the client...in that case, I think we should include the full details.